### PR TITLE
Provide an ordered and annotated config.yaml

### PR DESCRIPTION
**Pull request by wleese**
I found the config.yaml very, very hard to read.

Here's a copy of config.dev.yaml, with the following changes:

- split into sections: api, sender and sync
- annotate whatever I knew something of
- add commented out configuration I hadn't seen mentioned elsewhere
- consistently double # comments

Perhaps this belongs elsewhere. Just let me know.

### CODE REVIEW

File: configs/config.annotated.yaml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+## in the order listed here.
+role_lookups:
+  - user # support for 'role' being username
+  - mailing_list # mailing list config; see bottom of file
+  #- oncall # oncall service integration
+  - dummy # dummy role lookup used for testing purposes. Disable this in production.
+
+#oncall-api: https://oncall.dummy.com  # oncall url
+
+## TODO
```
Comment (dwang159) [inline configs/config.annotated.yaml]: TODO?

File: configs/config.annotated.yaml [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+#  # Max depth for recursing nested ldap lists.
+#  max_depth: 3
+#
+#  # If a list result returns at least this many results, don't support unrolling at runtime.
+#  max_unrolled_users: 100
+#
+#  # Every $n users, stop inserting into mysql to avoid crashing it. Comment these out to disable.
+#  user_add_pause_interval: 500
+#  user_add_pause_duration: 1
+
```
Comment (jrgp) [inline configs/config.annotated.yaml]: Should have a `iris-owa-sync` block here

File: configs/config.annotated.yaml [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+#oncall-api: https://oncall.dummy.com  # oncall url
+
+## TODO
+applications:
+ - dummy_app
+
+## Vendors provides a way to add support for new notification mechanisms
+vendors: []
+#- type: iris_slack
```
Comment (jrgp) [inline configs/config.annotated.yaml]: Should likely have the smtp example listed first (it seems missing)

File: configs/config.annotated.yaml [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+#
+#
+#metrics: influx
+#influxdb:
+#  connect:
+#    host: localhost
+#    port: 8086
+#    database: iris
+
+enable_gmail_oneclick: True
```
Comment (jrgp) [inline configs/config.annotated.yaml]: Please change this default from `True` to `False`, and add a comment linking to https://developers.google.com/gmail/markup/reference/one-click-action

File: configs/config.annotated.yaml (hunk 1 above)
Comment (wleese) [reply configs/config.annotated.yaml]: I currently have a poor understanding of the purpose of these 'applications'. I'll remove the TODO for now and hope to add a comment in the future

File: configs/config.annotated.yaml (hunk 3 above)
Comment (wleese) [reply configs/config.annotated.yaml]: I haven't seen any smtp configuration examples in the code base - did I miss something?

Comment (wleese) [reply configs/config.annotated.yaml]: Just took a quick look at the code and extracted the parameters.

File: configs/config.annotated.yaml [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+applications:
+ - dummy_app
+
+## Vendors provides a way to add support for new notification mechanisms
+vendors: []
+#- type: iris_smtp
+#  name: email
+#  smtp_server: 'smtp1.example.com'
+#  smtp_gateway: 'smtp2.example.com'
+#  from: 'foouser@example.com'
```
Comment (jrgp) [inline configs/config.annotated.yaml]: Can you add, double commented out, `autoscale: False` and `tasks_per_mx: 4`, with a comment describing how, if enabled, these will make a pool of persistent connections per each MX record for the given smtp_gateway?

Comment (wleese) [reply configs/config.annotated.yaml]: Not entirely sure if I understand the double commented out correctly, but I've followed your instructions as literally as I could.

**Comment by houqp** [comment]
Any reason not to just replace `config.dev.yaml` with this? It's much more maintainable.

**Comment by wleese** [comment]
@houqp I'm fine with that too. @dwang159 @jrgp agreed?

**Comment by houqp** [comment]
Easier to keep everything in sync if there is only one file to manage :)

**Comment by dwang159** [comment]
Closing in favor of #336

**Pull request closed without merging**